senza/manaus/cloudformation.py

```diff
@@ -124,6 +124,11 @@ def resources(self) -> Iterator:
         for resource in resources:
             resource_type = resource["ResourceType"]
             if resource_type == ResourceType.route53_record_set:
+                physical_resource_id = resource.get('PhysicalResourceId')
+                if physical_resource_id is None:
```

Comment:
Maybe we should just check if the status is `DELETE_COMPLETE`? Sounds more contextualised than we cannot check because there is no key found in the dict.